src/main/java/com/google/flourbot/entity/EntityModuleImplementation.java

```diff
@@ -91,7 +92,23 @@ public EntityModuleImplementation(DataStorage dataStorage) {
         String sheetAction = (String) actionData.get("sheetAction");
         String sheetUrl = (String) actionData.get("sheetUrl");
 
-        Action action = new SheetAppendAction(columnValue, sheetAction, sheetUrl);
+        Action action = null;
+
+        switch(sheetAction) {
+          case ("Sheet Append Action"):
+            action = new SheetAppendRowAction(columnValue, sheetAction, sheetUrl);
+            break;
+ 
+          case ("Read Row Action"):
+            action = new SheetReadRowAction(columnValue, sheetAction, sheetUrl);
+            break;
+
+          case ("Batch Action"):
```

Comment:
what is going on whith Batch Action? It is not safe to leave it like this. Throw java.lang.UnsupportedOperationException instead to make sure we come back to this code to fix it.